services/fdb/server/fdbmovedata.go

```diff
@@ -0,0 +1,170 @@
+   "License"); you may not use this file except in compliance
+   with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied.  See the License for the
+   specific language governing permissions and limitations
+   under the License.
+*/
+
+package server
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os/exec"
+	"sync"
+
+	"github.com/Snowflake-Labs/sansshell/services"
+	pb "github.com/Snowflake-Labs/sansshell/services/fdb"
+	"github.com/Snowflake-Labs/sansshell/services/util"
+	"github.com/go-logr/logr"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var (
+	// fdb_move_orchestrator binary location
+	FDBMoveOrchestrator     string
+	generateFDBMoveDataArgs = generateFDBMoveDataArgsImpl
+)
+
+type fdbmovedata struct {
+	mu  sync.Mutex
+	id  int64
+	cmd *exec.Cmd
+}
+
+func generateFDBMoveDataArgsImpl(req *pb.FDBMoveDataCopyRequest) ([]string, error) {
+	command := []string{FDBMoveOrchestrator}
+
+	var args []string
+	args = append(args, "--cluster", req.ClusterFile)
+	args = append(args, "--tenant-group", req.TenantGroup)
+	args = append(args, "--src-name", req.SourceCluster)
+	args = append(args, "--dst-name", req.DestinationCluster)
+	args = append(args, "--num-procs", fmt.Sprintf("%d", req.NumProcs))
+
+	command = append(command, args...)
+	return command, nil
+}
+
+func (s *fdbmovedata) FDBMoveDataCopy(ctx context.Context, req *pb.FDBMoveDataCopyRequest) (*pb.FDBMoveDataCopyResponse, error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	logger := logr.FromContextOrDiscard(ctx)
+	// The sansshell server should only run one copy command at a time
+	if !(s.cmd == nil) {
+		logger.Info("existing command already running. returning early")
+		earlyresp := &pb.FDBMoveDataCopyResponse{
+			Id: s.id,
+		}
+		return earlyresp, nil
+	}
+	s.id = rand.Int63()
+	command, err := generateFDBMoveDataArgs(req)
+	if err != nil {
+		return nil, err
+	}
+
+	// Add env vars for python and python bindings
+	cmd := exec.Command(command[0], command[1:]...)
+	// TODO: change to finalized locations when ready
+	cmd.Env = append(cmd.Environ(), "PYTHONPATH=/home/teleport-jfu/python")
+	cmd.Env = append(cmd.Environ(), "PATH=/opt/rh/rh-python36/root/bin")
+
+	logger.Info("executing local command", "cmd", cmd.String())
+	s.cmd = cmd
+	err = s.cmd.Start()
+	if err != nil {
+		s.cmd = nil
+		return nil, status.Errorf(codes.Internal, "error running fdbmovedata cmd (%+v): %v", command, err)
+	}
+
+	resp := &pb.FDBMoveDataCopyResponse{
+		Id: s.id,
+	}
+	return resp, nil
+}
+
+func (s *fdbmovedata) FDBMoveDataWait(ctx context.Context, req *pb.FDBMoveDataWaitRequest) (*pb.FDBMoveDataWaitResponse, error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
```

Comment:
We should probably release this lock when waiting on the command.

If we don't, any other concurrent rpcs for this will hang until the lock is released.